arraycontext/impl/pytato/compile.py

```diff
@@ -415,12 +415,23 @@ def _dag_to_transformed_pytato_prg(self, dict_of_named_arrays, *, prg_id=None):
                 prg_id, "pre_generate_loopy", pt_dict_of_named_arrays)
 
         with ProcessLogger(logger, f"generate_loopy for '{prg_id}'"):
+            import pyopencl as cl
+            dev = self.actx.context.devices[0]
+            target = None
+            if (dev.type & cl.device_type.GPU
+                    and cl.characterize.has_coarse_grain_buffer_svm(dev)):
+                limit = dev.max_parameter_size
+                # Leave some extra space since our sizes are estimates
+                target = lp.PyOpenCLTarget(limit_arg_size_nbytes=limit//2)
```

Comment:
Hmm, upon second thought: We can only pass this if we're sure that the memory allocated is actually SVM. So this has to get involved with memory pool creation.

Reply:
Should we do this here or as part of https://github.com/inducer/loopy/pull/642 ?

Reply:
Done in https://github.com/inducer/arraycontext/pull/189/commits/faba326cd2142d4072c1132c523e6c82553621df and https://github.com/inducer/arraycontext/pull/189/commits/61038f97575019bc95903995f778ff1283a2c014